mito-ds/mito
# Further Dash Fixes

# Description

This is a reasonably sprawling PR that fixes a bunch of issues with our Dash integration:
1. Moves dropdowns from `absolute` positioning to `fixed` positioning, so that positioning works the same across JupyterLab and Streamlit as well as Dash. The issue was that JupyterLab never scrolls the actual window, nor does streamlit (as it's in an iframe) -- so we never previously had to deal with this case. 
2. Makes the `@mito_callback` work in more cases, by doing 0 manipulation of the Input and State arguments to the `@callback` function (which has a whole bunch of ways you can [pass these](https://dash.plotly.com/flexible-callback-signatures)). 
3. Makes `@mito_callback` give you an error message when you use `@callback` instead of it. Now, it actually tells you what is wrong -- so that you have a hope of fixing it without being hopelessly confused. We could write some more docs here, as well.
4. Makes it so selection changes don't also trigger callbacks for `spreadsheet_result` to run. This was the source of performance problems, and generally it makes a ton of sense to be able to subscribe to each separately. 
5. Fixes some CSS issues: sheet tab coloring in Dash.
6. Adds the ability to pass a theme to a Dash component -- so our demo apps can look better. This is a bit buggy with multiple mitosheets on one page, when one is themed and another is not, but I think it's ok.

# Testing

1. Test dropdowns place themselves well on JupyterLab, Streamlit, and Dash.
2. Try the new `@mito_callback` decorator -- see if you can pass lists of Inputs and States (and make sure that it runs properly in this cases)
3. Use `@callback` and see the error message you get when using it with a Spreadsheet input - see if it's helpful.
4. Check the new dash usage, and check that selection changes trigger selection change callbacks, and that spreadsheet result changes trigger new spreadsheet result callbacks
5. Check sheet coloring. 
6. Check dash usage as well, see passed theme.

# Documentation

Once all this is merged in, I'm going to take another pass at the Dash documentation!
